internal/index/manager/manager.go

```diff
@@ -821,6 +829,27 @@ func (mgr *Manager) getIndexesCopy(start int) ([]*index.Reader, indexReleaser) {
 	return indexes, mgr.lock(indexes)
 }
 
+func (mgr *Manager) SetClientConfig(config ClientConfig) ClientConfig {
+	mgr.clientConfig = config
+	return mgr.ClientConfig()
```

Comment:
Just return `mgr.clientConfig` to avoid a deadlock with this job new waiting for the `ClientConfig()` job. In the future we could allow partial updates when there are more config options in the struct.